### ISSUE TITLE
Bacalhau only supports the use of 1 GPU

### ISSUE THREAD

**Issue by philwinder** [bacalhau-project/bacalhau#1665]
Only 1 GPU can be used at a time: https://github.com/filecoin-project/bacalhau/blob/dev/pkg/executor/docker/executor.go#L225
The [Docker API expects](https://docs.docker.com/config/containers/resource_constraints/#expose-gpus-for-use) you to specify the exact GPU you want to run on by ID or by device number.

At the time of writing this code I didn’t know how to or whether we wanted to support maintaining the state of what GPUs are in use by that particular node. Maybe there’s a way now, but that was considered future work. Hence the todo.

In summary, only GPU 0 can be used and the docs are correct.

In the future, we need to support both running multiple jobs on multiple GPUs and single jobs on multiple GPUs. But this is very tied to how GPU jobs are specified and state is maintained so there's probably some dependent work that is required first. See the main GPU improvement issue for this.

**Comment by lukemarsden** [comment]
We need this to support compute providers who are showing up with multi-GPU servers
